entrypoint.sh

```diff
@@ -2,153 +2,133 @@
 # shellcheck disable=SC2046,SC2001,SC2086
 set -e
 
-if [ -z "$OP_CONNECT_TOKEN" ] || [ -z "$OP_CONNECT_HOST" ]; then
-  echo "\$OP_CONNECT_TOKEN and \$OP_CONNECT_HOST must be set"
-  exit 1
-fi
+readonly CONNECT="CONNECT"
+readonly SERVICE_ACCOUNT="SERVICE_ACCOUNT"
 
+auth_type=$CONNECT
 managed_variables_var="OP_MANAGED_VARIABLES"
-IFS=',' read -r -a managed_variables <<< "$(printenv $managed_variables_var)"
+IFS=','
 
 # Unset all secrets managed by 1Password if `unset-previous` is set.
-if [ "$INPUT_UNSET_PREVIOUS" == "true" ]; then
-  echo "Unsetting previous values..."
+unset_prev_secrets() {
+  if [ "$INPUT_UNSET_PREVIOUS" == "true" ]; then
+    echo "Unsetting previous values..."
 
-  # Find environment variables that are managed by 1Password.
-  for env_var in "${managed_variables[@]}"; do
-    echo "Unsetting $env_var"
-    unset $env_var
+    # Find environment variables that are managed by 1Password.
+    for env_var in "${managed_variables[@]}"; do
+      echo "Unsetting $env_var"
+      unset $env_var
 
-    echo "$env_var=" >> $GITHUB_ENV
+      echo "$env_var=" >> $GITHUB_ENV
 
-    # Keep the masks, just in case.
-  done
-
-  managed_variables=()
-fi
-
-curl_headers=(-H "Content-Type: application/json" -H "Authorization: Bearer $OP_CONNECT_TOKEN")
+      # Keep the masks, just in case.
+    done
 
-# Iterate over environment varables to find 1Password references, load the secret values, 
-# and make them available as environment variables in the next steps.
-IFS=$'\n'
-for possible_ref in $(printenv | grep "=op://" | grep -v "^#"); do
-  env_var=$(echo "$possible_ref" | cut -d '=' -f1)
-  ref=$(printenv $env_var)
-
-  if [[ ! $ref == "op://"* ]]; then
-    echo "Not really a reference: $ref"
-    continue
+    managed_variables=()
   fi
-
-  path=$(echo $ref | sed -e "s/^op:\/\///")
-  if [ $(echo "$path" | tr -cd '/' | wc -c) -lt 2 ]; then
-    echo "Expected path to be in format op://<vault>/<item>[/<section>]/<field>: $ref"
-    continue
+}
+
+# Install op-cli
+install_op_cli() {
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    curl -sSfLo op.zip "https://cache.agilebits.com/dist/1P/op2/pkg/v2.6.0-beta.06/op_linux_amd64_v2.6.0-beta.06.zip"
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    curl -sSfLo op.zip "https://cache.agilebits.com/dist/1P/op2/pkg/v2.6.0-beta.06/1password-cli_v2.6.0-beta.06_darwin_amd64.tar.gz"
```

Comment:
Does it really work this way? Downloading a `.tar.gz` and save it as a `.zip`...I assume a different type of archive works differently, and changing the file extension may lead to some unexpected surprises.

Reply:
changed the way to correctly extract `.tar.gz`